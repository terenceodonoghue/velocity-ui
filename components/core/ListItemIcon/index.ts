export { default } from './ListItemIcon';
export type { ListItemIconProps } from './ListItemIcon.types';
