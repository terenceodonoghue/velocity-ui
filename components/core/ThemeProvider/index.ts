export { default, defaultTheme } from './ThemeProvider';
export type { Theme } from './ThemeProvider.types';
