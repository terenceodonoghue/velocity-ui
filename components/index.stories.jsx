/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { storiesOf } from '@storybook/react';
import faker from 'faker';
import {
  Avatar, Button, Container, Input, Page,
} from 'components';
import * as S from './index.styles';

storiesOf('Basics|Style Guide', module).add('Components', () => (
  <S.Wrapper>
    <Page.AppBar>
      <Avatar.Round
        alt="Avatar"
        src={faker.image.avatar()}
        height="40"
        width="40"
      />
      <Button.Text width="40px">
        <img alt="Mail" src="./mail.svg" />
      </Button.Text>
      <Button.Text width="40px">
        <img alt="Notifications" src="./notifications.svg" />
      </Button.Text>
    </Page.AppBar>
    <Container.Card title="Avatars">
      <S.Stories>
        <Avatar.Round src={faker.image.avatar()} height="48" width="48" />
      </S.Stories>
    </Container.Card>
    <Container.Card title="Buttons">
      <S.Stories>
        <Button.Contained>
        Button Text
        </Button.Contained>
        <Button.Outlined>
        Button Text
        </Button.Outlined>
        <Button.Text>
        Button Text
        </Button.Text>
        <Button.Fab>
          <img alt="Fab" src="./plus.svg" />
        </Button.Fab>
      </S.Stories>
    </Container.Card>
    <Container.Card title="Inputs">
      <S.Stories>
        <Input.Label>
            Label
          <Input.Text
            type="text"
            placeholder="Placeholder"
          />
        </Input.Label>
        <Input.Switch />
        <Input.Slider
          min={0}
          max={100}
        />
      </S.Stories>
    </Container.Card>
  </S.Wrapper>
));
