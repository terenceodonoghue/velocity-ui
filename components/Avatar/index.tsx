/** @jsx jsx */
import { jsx } from '@emotion/core';
import { FunctionComponent, ImgHTMLAttributes } from 'react';
import * as css from './Avatar.styles';

interface AvatarProps extends ImgHTMLAttributes<HTMLImageElement> {
  alt: string;
  round?: boolean;
}

const Avatar: FunctionComponent<AvatarProps> = ({ alt, round, ...props }) => (
  <img alt={alt} css={round ? css.round : css.square} {...props} />
);

export default Avatar;
