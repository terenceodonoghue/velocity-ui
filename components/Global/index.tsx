/** @jsx jsx */
import { css, Global, jsx } from '@emotion/core';
import { FunctionComponent, ReactElement } from 'react';
import { ThemeProvider as EmotionProvider } from 'emotion-theming';
import { Theme } from '~/types';

export const theme: Theme = {
  colors: {
    black: '#2e384d',
    blueyGrey: '#8798ad',
    blue: '#2e5bff',
    blueDark: '#0036fa',
    green: '#33ac2e',
    lightBlueGrey: '#bfc5d2',
    purple: '#8c54ff',
    red: '#d63649',
    teal: '#00c1d4',
    white: '#fff',
    yellow: '#f7c137',
  },
  effects: {
    transition: 'all 0.1s linear',
  },
  typography: {
    baseFontSize: '15px',
    baseFontFamily: 'Rubik, sans-serif',
    fontWeightLight: 300,
    fontWeightRegular: 400,
    fontWeightMedium: 500,
    fontWeightBold: 700,
  },
};

export const Styles = (): ReactElement => (
  <Global
    styles={css`
      :any-link {
        text-decoration: none;
      }

      * {
        box-sizing: border-box;
      }

      body {
        margin: 0;
        background-color: rgba(46, 91, 255, 0.08);
        color: ${theme.colors.black};
        font-family: ${theme.typography.baseFontFamily};
        font-size: ${theme.typography.baseFontSize};
      }

      main {
        margin-left: 80px;
        max-width: 1360px;
        padding: 64px 110px;
      }
    `}
  />
);

export const ThemeProvider: FunctionComponent = ({ children }) => (
  <EmotionProvider theme={theme}>{children}</EmotionProvider>
);
