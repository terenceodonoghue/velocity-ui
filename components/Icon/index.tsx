/** @jsx jsx */
import { FunctionComponent, ReactElement } from 'react';
import { jsx } from '@emotion/core';
import { useTheme } from 'emotion-theming';
import { IconProps, Theme } from '~/types';
import * as svg from './svg';
import icon from './Icon.styles';

const withTheme = (
  BaseIcon: FunctionComponent<IconProps>,
): FunctionComponent<IconProps> => ({
  fill,
  size = 24,
  title,
  ...props
}): ReactElement => {
  const theme = useTheme<Theme>();

  return (
    <BaseIcon
      {...props}
      css={icon(size)}
      fill={fill}
      theme={theme}
      title={title}
    />
  );
};

export const Analytics = withTheme(svg.Analytics);
export const Chat = withTheme(svg.Chat);
export const Dashboard = withTheme(svg.Dashboard);
export const Map = withTheme(svg.Map);
export const Messages = withTheme(svg.Messages);
export const Notifications = withTheme(svg.Notifications);
export const Service = withTheme(svg.Service);
export const Settings = withTheme(svg.Settings);
export const Vehicles = withTheme(svg.Vehicles);
