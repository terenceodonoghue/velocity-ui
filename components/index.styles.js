import styled from '@emotion/styled';
import { Container } from 'components';

export const Components = styled.div`
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  margin: -12px;

  & > * {
    margin: 12px;
  }
`;

export const Story = styled(Container.Card)`
  margin: 24px 0;
  padding: 32px 24px;
`;
