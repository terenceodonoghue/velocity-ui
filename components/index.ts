import * as Button from './Button';
import * as Global from './Global';
import * as Icon from './Icon';
import * as Input from './Input';
import * as Layout from './Layout';

export { default as Avatar } from './Avatar';
export { Button, Global, Icon, Input, Layout };
