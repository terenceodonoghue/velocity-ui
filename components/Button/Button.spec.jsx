/** @jsx jsx */
import { jsx } from '@emotion/core';
import { Button } from '~/components';

describe('Button', () => {
  describe('Contained', () => {
    it('matches snapshot', () => {
      const tree = renderer
        .create(withTheme(<Button.Contained>Button Text</Button.Contained>))
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Outlined', () => {
    it('matches snapshot', () => {
      const tree = renderer
        .create(withTheme(<Button.Outlined>Button Text</Button.Outlined>))
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Text', () => {
    it('matches snapshot', () => {
      const tree = renderer
        .create(withTheme(<Button.Text>Button Text</Button.Text>))
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Fab', () => {
    it('matches snapshot', () => {
      const tree = renderer
        .create(withTheme(<Button.Fab>+</Button.Fab>))
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
