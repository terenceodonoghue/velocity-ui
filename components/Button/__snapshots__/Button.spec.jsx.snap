// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Contained matches snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  padding: 0;
  font-size: 15px;
  font-weight: 500;
  cursor: pointer;
  -webkit-transition: all 0.2s linear;
  transition: all 0.2s linear;
  border-radius: 4px;
  height: 40px;
  width: 200px;
  background-color: #2e5bff;
  color: #fff;
}

.emotion-0:disabled {
  opacity: 0.3;
  pointer-events: none;
}

.emotion-0:focus {
  outline: 0;
}

.emotion-0:enabled:hover {
  background-color: #0036fa;
}

<button
  className="emotion-0"
  type="button"
>
  Button Text
</button>
`;

exports[`Button Fab matches snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  padding: 0;
  font-size: 15px;
  font-weight: 500;
  cursor: pointer;
  -webkit-transition: all 0.2s linear;
  transition: all 0.2s linear;
  border-radius: 50%;
  height: 70px;
  width: 70px;
  background-color: #2e5bff;
  color: #fff;
}

.emotion-0:disabled {
  opacity: 0.3;
  pointer-events: none;
}

.emotion-0:focus {
  outline: 0;
}

.emotion-0:enabled:hover {
  background-color: #0036fa;
}

<button
  className="emotion-0"
  type="button"
>
  +
</button>
`;

exports[`Button Outlined matches snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  padding: 0;
  font-size: 15px;
  font-weight: 500;
  cursor: pointer;
  -webkit-transition: all 0.2s linear;
  transition: all 0.2s linear;
  border-radius: 4px;
  height: 40px;
  width: 200px;
  background: rgba(46,91,255,0.2);
  color: #2e5bff;
}

.emotion-0:disabled {
  opacity: 0.3;
  pointer-events: none;
}

.emotion-0:focus {
  outline: 0;
}

.emotion-0:enabled:hover {
  background-color: rgba(46,91,255,0.3);
}

<button
  className="emotion-0"
  type="button"
>
  Button Text
</button>
`;

exports[`Button Text matches snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  padding: 0;
  font-size: 15px;
  font-weight: 500;
  cursor: pointer;
  -webkit-transition: all 0.2s linear;
  transition: all 0.2s linear;
  border-radius: 4px;
  height: 40px;
  width: 200px;
  background: none;
  color: #2e5bff;
}

.emotion-0:disabled {
  opacity: 0.3;
  pointer-events: none;
}

.emotion-0:focus {
  outline: 0;
}

.emotion-0:enabled:hover {
  background-color: rgba(46,91,255,0.1);
}

<button
  className="emotion-0"
  type="button"
>
  Button Text
</button>
`;
